client/my-sites/patterns/components/pattern-preview/index.tsx

```diff
@@ -46,6 +51,9 @@ type PatternPreviewProps = {
 	isResizable?: boolean;
 	pattern: Pattern | null;
 	viewportWidth?: number;
+	category: string;
+	view: PatternView;
+	patternTypeFilter: PatternTypeFilter;
```

Comment:
```suggestion
	category?: string;
	view?: PatternView;
	patternTypeFilter?: PatternTypeFilter;
```
Let's make these properties optional, so we don't have to worry about updating `client/my-sites/patterns/components/category-gallery/client.tsx` to please Typescript.

Reply:
Yeah, I was kind of on the fence with this one. On the one hand I didn't love passing unneeded props as you described, and on the other I didn't want to introduce more TS errors by making props optional.

We actually eliminated the need for `view` entirely in #88878, and after your nudge here I've made the remaining two props optional. I only wound up needing to navigate one type check, which feels good to me! (8641913d8a)